app/app.py

```diff
@@ -187,9 +188,16 @@ def translate():
         if char_limit != -1:
             q = q[:char_limit]
 
+        original_source_lang = source_lang
+        if source_lang == 'auto':
+            source_lang = detect(q)
+
         src_lang = next(iter([l for l in languages if l.code == source_lang]), None)
         tgt_lang = next(iter([l for l in languages if l.code == target_lang]), None)
-
+        
+        if src_lang is None and original_source_lang == 'auto':
+            return jsonify({"translatedText": "Detected language not supported (" + source_lang + ")" })
```

Comment:
Wouldn't it be better to throw an http error code here? So you could also show a matching error message in the frontend or handle it in an application that is using this api.
Otherwise, there is a risk that a user will see this as a translation because a successful 200 is coming.

Reply:
No, because LibreTranslate try to translate at every key push, so you are getting a lot of languages that are not supported before finishing typing. It was actually the main issue to take of with the UI. But this should be improved by asking for the list of most probable langs and using the first that we support

```
from langdetect import detect_langs
>>> detect_langs("Otec matka syn.")
[sk:0.572770823327, pl:0.292872522702, cs:0.134356653968]
```